sdktests/server_side_persistence_dynamodb.go

```diff
@@ -0,0 +1,246 @@
+		ComparisonOperator: aws.String("EQ"),
+		AttributeValueList: []*dynamodb.AttributeValue{{
+			S: aws.String(addPrefix(prefix, key)),
+		}},
+	}
+
+	// Read in all the old keys first
+	query := &dynamodb.QueryInput{
+		TableName:      aws.String(tableName),
+		ConsistentRead: aws.Bool(true),
+		KeyConditions: map[string]*dynamodb.Condition{
+			tablePartitionKey: &condition,
+		},
+	}
+
+	response, err := d.dynamodb.Query(query)
+	if err != nil {
+		return err
+	}
+
+	for _, item := range response.Items {
+		itemKey := item[tableSortKey].String()
+		unusedKeys[itemKey] = true
+	}
+
+	requests := make([]*dynamodb.WriteRequest, 0)
+
+	for k, v := range data {
+		var versioned struct {
+			Version int `json:"version"`
+		}
+		if err := json.Unmarshal([]byte(v), &versioned); err != nil {
+			return err
+		}
+		requests = append(requests, &dynamodb.WriteRequest{
+			PutRequest: &dynamodb.PutRequest{
+				Item: map[string]*dynamodb.AttributeValue{
+					tablePartitionKey: {S: aws.String(addPrefix(prefix, key))},
+					tableSortKey:      {S: aws.String(k)},
+					itemJSONAttribute: {S: aws.String(v)},
+					versionAttribute:  {N: aws.String(strconv.Itoa(versioned.Version))},
+				},
+			},
+		})
+		delete(unusedKeys, k)
+	}
+
+	for k := range unusedKeys {
+		if k == initedKey {
+			continue
+		}
+		delKey := map[string]*dynamodb.AttributeValue{
+			tablePartitionKey: {S: aws.String(addPrefix(prefix, key))},
+			tableSortKey:      {S: aws.String(k)},
+		}
+		requests = append(requests, &dynamodb.WriteRequest{
+			DeleteRequest: &dynamodb.DeleteRequest{Key: delKey},
+		})
+	}
+
+	// Now set the special key that we check in InitializedInternal()
+	requests = append(requests, &dynamodb.WriteRequest{
+		PutRequest: &dynamodb.PutRequest{Item: map[string]*dynamodb.AttributeValue{
+			tablePartitionKey: {S: aws.String(addPrefix(prefix, initedKey))},
+			tableSortKey:      {S: aws.String(initedKey)},
+		}},
+	})
+
+	if err := batchWriteRequests(context.Background(), d.dynamodb, tableName, requests); err != nil {
+		// COVERAGE: can't cause an error here in unit tests because we only get this far if the
+		// DynamoDB client is successful on the initial query
+		return fmt.Errorf("failed to write %d items(s) in batches: %s", len(requests), err)
+	}
+
+	return nil
+}
+
+// batchWriteRequests executes a list of write requests (PutItem or DeleteItem)
+// in batches of 25, which is the maximum BatchWriteItem can handle.
+func batchWriteRequests(
+	context context.Context,
+	client *dynamodb.DynamoDB,
+	table string,
+	requests []*dynamodb.WriteRequest,
+) error {
+	for len(requests) > 0 {
+		batchSize := int(math.Min(float64(len(requests)), 25))
+		batch := requests[:batchSize]
+		requests = requests[batchSize:]
+
+		_, err := client.BatchWriteItem(&dynamodb.BatchWriteItemInput{
+			RequestItems: map[string][]*dynamodb.WriteRequest{table: batch},
+		})
+		if err != nil {
+			// COVERAGE: can't simulate this condition in unit tests because we will only get this
```

Comment:
Maybe this comment does apply, but the coverage suppression probably doesn't.

(Though I am not 100% sure it makes sense in its original location, which was also a test.)